presto-spi/src/main/java/com/facebook/presto/spi/PageSourceOptions.java

```diff
@@ -100,8 +100,8 @@ public FilterFunction(int[] inputChannels, int initialCost)
 
     public static class ErrorSet
     {
-        int positionCount;
-        private Throwable[] errors;
```

Comment:
Do we need to care in `ErrorSet` that it's just `RuntimeException`?  I.e. can we just change what we catch?

Reply:
I think we need to. Otherwise, we'll have to return Throwable in getFirstError or add some ugly cast + test logic.